controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java

```diff
@@ -114,7 +117,9 @@ private void checkIntegration(Integration integration) {
         Optional<Integration.Status> current = integration.getCurrentStatus();
         if (desired.isPresent() && !current.equals(desired)) {
             WorkflowHandler workflowHandler = handlers.get(desired.get());
-            if (workflowHandler != null) {
+
+            String scheduledKey = "" + integration.getDesiredStatus() + ":" + integration.getId().get();
+            if (workflowHandler != null && !scheduledChecks.containsKey(scheduledKey)) {
```

Comment:
Could this check be put directly within enqueue() (at the beginning) so that it doesn't depend on the order the methods are called, and its 'selfcontained' (e.g. scheduledChecks is only known to and used by `enqueue()`. The key needs to be calculated only once, too.